### PR TITLE
Refactor oversized method handle_forest_behaviour_event 

### LINKED ISSUE #2220: Refactor oversized method handle_forest_behaviour_event
**Issue summary**

As follow up of https://github.com/ChainSafe/forest/pull/2154#discussion_r1025141261

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #2220

**Other information and links**

### CODE REVIEW

**Review by hanabi1224 — APPROVED** [review]
LGTM

File: node/forest_libp2p/src/service.rs [hunk 1]
```diff
@@ -410,271 +410,292 @@ async fn handle_network_message<P: StoreParams>(
...
+    }
+}
+
+async fn handle_chain_exchange_event<DB, P: StoreParams>(
+    ce_event: RequestResponseEvent<ChainExchangeRequest, ChainExchangeResponse>,
+    db: &Arc<ChainStore<DB>>,
+    network_sender_out: &flume::Sender<NetworkEvent>,
+    cx_request_table: &mut HashMap<
+        RequestId,
+        futures::channel::oneshot::Sender<Result<ChainExchangeResponse, RequestResponseError>>,
```
Comment (lemmih) [inline node/forest_libp2p/src/service.rs:686]:
```suggestion
        OneShotSender<Result<ChainExchangeResponse, RequestResponseError>>,
```

Comment (lemmih) [reply node/forest_libp2p/src/service.rs:686]: There are several of these. You can update all of them.

File: node/forest_libp2p/src/service.rs [hunk 2]
```diff
@@ -716,44 +737,105 @@ async fn handle_forest_behaviour_event<DB, P: StoreParams>(
...
+    genesis_cid: &Cid,
+    network_sender_out: &flume::Sender<NetworkEvent>,
+    hello_request_table: &mut HashMap<
+        RequestId,
+        futures::channel::oneshot::Sender<Result<HelloResponse, RequestResponseError>>,
+    >,
+    cx_request_table: &mut HashMap<
+        RequestId,
+        futures::channel::oneshot::Sender<Result<ChainExchangeResponse, RequestResponseError>>,
+    >,
```
Comment (lemmih) [inline node/forest_libp2p/src/service.rs:795]: The type signatures for `hello_request_table` and `cx_request_table` are repeated three times. Would probably be worth it to use a type synonym.

File: node/forest_libp2p/src/service.rs [hunk 3]
```diff
@@ -716,44 +737,105 @@ async fn handle_forest_behaviour_event<DB, P: StoreParams>(
...
+    outgoing_bitswap_query_ids: &mut HashMap<libp2p_bitswap::QueryId, Cid>,
+    cx_response_tx: Sender<(
```
Comment (lemmih) [inline node/forest_libp2p/src/service.rs:797]: Either use `flume::Sender` everywhere or `Sender` everywhere. I'm leaning towards dropping the `flume` part.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: node/forest_libp2p/src/service.rs (hunk 3 above)
Comment (tyshko5) [reply node/forest_libp2p/src/service.rs:797]: But `flume::Sender` is an mpmc channel and `OneShot`(or `Sender`) is spsc channel. Wouldn't that break the logic?

Comment (lemmih) [reply node/forest_libp2p/src/service.rs:797]: I didn't mean you should replace `OneShotSender` with `Sender`. My point is that you're using both `flume::Sender` and `Sender` which refers to the same type. This should be made more consistent by either always using the qualified `flume::Sender` or always using the unqualified `Sender`.

Comment (tyshko5) [reply node/forest_libp2p/src/service.rs:785]: Done

File: node/forest_libp2p/src/service.rs (hunk 2 above)
Comment (tyshko5) [reply node/forest_libp2p/src/service.rs]: Done

File: node/forest_libp2p/src/service.rs (hunk 1 above)
Comment (tyshko5) [reply node/forest_libp2p/src/service.rs]: Done

**Review by lemmih — APPROVED** [review]

**Pull request merged**